deb/openmediavault/usr/share/openmediavault/mkconf/samba.d/40recyclebin

```diff
@@ -69,9 +69,9 @@ done
 # Exit if 'recyclemaxage' is zero.
 [ "\${amin}" -eq "0" ] && exit 0
 # Exit if another job is running.
-if ! mkdir "${lockfile}" &>/dev/null; then
-	exit 0
-fi
+[ -e ${lockfile} ] && exit 0
```

Comment:
The initial solution has been choosen because mkdir is atomic. With your implementation the creation and checking of the lock file is not atomic anymore because it is splitted into separate operations.

Reply:
The problem should be solved if the trap is relocated before the mkdir command. I have to check that.

Reply:
With actual solution script never do its work of cleaning recycle because exit always with exit status 0 before, first time creating "lockfile" and after because exist, move trap before delete the "lockfile" on exit but still exit without execute the clean, the only working solution is split create and check "lockfile" (I not know way to make it working without split)